gui/src/app/Stanc/stancWorker.ts

```diff
@@ -0,0 +1,48 @@
+import { StancFunction, StancRequestMessage, Replies, Requests } from "./Types";
+import rawStancJS from "./stanc.js?raw";
+
+let stanc: undefined | StancFunction;
+try {
+  // stanc.js code is not a module, so most nice options for loading are unavailable
+  eval(rawStancJS);
+  // stanc.js also detects if it is running under node, which makes testing annoying
+  if (typeof module !== "undefined") {
+    // node (default vitest setup)
+    stanc = module.exports.stanc;
+  } else {
+    // browser
+    stanc = (globalThis as any).stanc;
+  }
+  console.log("loaded stanc.js");
+} catch (e) {
+  console.error("Failed to load stanc.js");
+  console.error(e);
+}
+
+self.onmessage = (e: MessageEvent<StancRequestMessage>) => {
+  const { purpose, name, code } = e.data;
+
+  if (!stanc) {
+    self.postMessage({ error: "stanc.js not loaded!" });
+    return;
+  }
+
+  const args = [`filename-in-msg=${name}`, "auto-format", "max-line-length=78"];
+
+  const output = stanc(name, code, args);
```

Comment:
Would be nice to describe a little bit more the interface of the `stanc` entity (esp. since it's loaded as a code blob). I think I get it from looking at the usage, but I am insecure when dealing with ambiguity